Snowflake.Data.Tests/UnitTests/Tools/TimeoutHelperTest.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using Snowflake.Data.Core.Tools;
+
+namespace Snowflake.Data.Tests.UnitTests.Tools
+{
+    [TestFixture]
+    public class TimeoutHelperTest
+    {
+        [Test]
+        [TestCaseSource(nameof(InfinityTimeouts))]
+        public void TestInfinity(TimeSpan infiniteTimeout)
```

Comment:
I don't like the plural form because the test method is for one case only (although it is run for every case). So it should be rather TestInfiniteValue but if so, then TestInfinity is shorter.